string mask_filename
---
bool success
